new-site/docs-site/content/0.19.0/guide/typesense-dynamodb.md

```diff
@@ -0,0 +1,297 @@
+
+![DynamoDB Triggers](~@guide-images/typesense-dynamodb/dynamodb-typesense-integration-3.png)
+
+Then now you need to create a "Lambda Execution Role" i.e give permission for your function. Now, head over to IAM Roles section and create a new role with three main permissions:
+
+* AmazonDynamoDBFullAccess
+* AmazonDynamoDBFullAccesswithDataPipeline
+* AWSLambdaBasicExecutionRole
+
+![IAM Execution Role](~@guide-images/typesense-dynamodb/dynamodb-typesense-integration-4.png)
+
+Now, attach this execution role to the trigger Lambda funtion of DynamoDB.
+
+
+## Create Lambda Execution Role (only for Command Line)
+
+Create a file named ```trust-relationship.json``` with the following contents.
+```json
+{
+   "Version": "2012-10-17",
+   "Statement": [
+     {
+       "Effect": "Allow",
+       "Principal": {
+         "Service": "lambda.amazonaws.com"
+       },
+       "Action": "sts:AssumeRole"
+     }
+   ]
+}
+```
+
+Then, create execute the following command
+
+```bash
+aws iam create-role --role-name TypesenseLambdaRole \
+    --path "/service-role/" \
+    --assume-role-policy-document file://trust-relationship.json
+```
+
+Now, create ```role-policy.json``` with the following contents. (Replace ```accountID``` and ```region```)
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "lambda:InvokeFunction",
+            "Resource": "arn:aws:lambda:region:accountID:function:typesense-indexing*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogGroup",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+            "Resource": "arn:aws:logs:region:accountID:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "dynamodb:DescribeStream",
+                "dynamodb:GetRecords",
+                "dynamodb:GetShardIterator",
+                "dynamodb:ListStreams"
+            ],
+            "Resource": "arn:aws:dynamodb:region:accountID:table/typesense/stream/*"
+        },
+    ]
+}
+```
+
+The policy has three statements that allow TypesenseLambdaRole to do the following:
+
+* Run a Lambda function ```typesense-indexing```. You create the function later in this tutorial.
+* Access Amazon CloudWatch Logs. The Lambda function writes diagnostics to CloudWatch Logs at runtime.
+* Read data from the DynamoDB stream for ```typesense```.
+
+Now, we are going to attach the above roles to our IAM execution role which we have created
+
+```bash
+aws iam put-role-policy --role-name TypesenseLambdaRole \
+    --policy-name TypesenseLambdaRolePolicy \
+    --policy-document file://role-policy.json
+```
+
+## Importing ```typesense``` python package in Lambda function
```

Comment:
Since using Python with Lambda requires additional setup work, it takes focus away from the primary purpose of this article. Could we instead use Javascript here, which doesn't require the additional setup steps?